src/bitcoin-tx.cpp

```diff
@@ -233,9 +266,78 @@ static void MutateTxAddOutAddr(CMutableTransaction& tx, const string& strInput)
     CBitcoinAddress addr(strAddr);
     if (!addr.IsValid())
         throw runtime_error("invalid TX output address");
+    CScript scriptPubKey; // The scriptPubKey in the txOut
+    CScript script = GetScriptForDestination(addr.Get());
 
-    // build standard output script via GetScriptForDestination()
-    CScript scriptPubKey = GetScriptForDestination(addr.Get());
+    if (!paytowitnesspubkeyhash) {
+        // This is a pay to pub key hash. Build standard output script via
+        // GetScriptForDestination()
+        scriptPubKey = script;
+    }
+    else {
+        // This is a pay to witness pub key hash. Build the scriptPubKey.
+        scriptPubKey = GetScriptForWitness(script);
+    }
+
+    // construct TxOut, append to transaction output list
+    CTxOut txout(value, scriptPubKey);
+    tx.vout.push_back(txout);
+}
+
+static void MutateTxAddMultiSig(CMutableTransaction& tx, const string& strInput, bool paytoscripthash, bool paytowitnessscripthash)
+{
+    // Seperate into VALUE:REQUIRED:ADDRESSES:ADDRESS1:ADDRESS2:....
+    std::vector<std::string> vStrInputParts;
+    boost::split(vStrInputParts, strInput, boost::is_any_of(":"));
+
+    // Check that there are enough parametners
+    if (vStrInputParts.size()<3)
+        throw runtime_error("Not enough multisig parameters");
+
+    // Extract and validate VALUE
+    string strValue = vStrInputParts[0];
+    CAmount value;
+    if (!ParseMoney(strValue, value))
+        throw runtime_error("invalid TX output value");
+
+    // Extract REQUIRED
+    uint required = std::stoul(vStrInputParts[1]);
+
+    // Extract NUMKEYS
+    uint numkeys = std::stoul(vStrInputParts[2]);
+
+    // Validate there are the correct number of pubkeys
+    if (vStrInputParts.size() != numkeys + 3)
+        throw runtime_error("incorrect number of multisig pubkeys");
+
+    // extract and validate PUBKEYs
+    std::vector<CPubKey> pubkeys;
+    for(uint pos = 1; pos <= numkeys; pos++) {
+        // extract and validate PUBKEY
+        CPubKey pubkey(ParseHex(vStrInputParts[pos + 2]));
+        if (!pubkey.IsFullyValid())
+            throw runtime_error("invalid TX output pubkey");
+        pubkeys.push_back(pubkey);
+    }
+
+    CScript scriptPubKey; // The scriptPubKey in the txOut
+    CScript script = GetScriptForMultisig(required, pubkeys);
+
+    if (paytoscripthash) {
+        // Get the address for the redeem script, then call
+        // GetScriptForDestination() to construct a P2SH scriptPubKey.
+        CBitcoinAddress addr(script);
+        scriptPubKey = GetScriptForDestination(addr.Get());
+    }
+    else if (paytowitnessscripthash) {
```

Comment:
Yes, it's possible to create a P2WSH wrapped in a P2SH, but I haven't implemented that in bitcoin-tx. If you think that's going to be useful, you could add separate commands for:

- multisig in P2WSH wrapped in P2SH
- any script in P2WSH wrapped in P2SH
- P2WPK wrapped in P2SH

I don't think there's much use case for this, but feel free to create an additional PR on top of this if you think it'd be useful.